Electron/Core/RoveProtocol/Rovecomm.ts

```diff
@@ -126,35 +134,120 @@ function parse(packet: Buffer): void {
   }
 }
 
-function TCPListen(socket: Socket) {
+function TCPParseWrapper(socket: TCPSocket) {
+  /*
+   * Takes in an object of the TCPSocket type defined in the interface TCPSocket at the top of this file
+   * Iterates while there is still at least five bytes in the Deque in the TCPSocket, parsing one RC Packet at a time
+   * Returns when there is either less than 5 bytes in the Deque or not a complete packet in the Deque
+   */
+  // While the Deque contains at least a header to allow parsing control packets
+  while (socket.RCDeque.length >= 5) {
+    const dataCount = socket.RCDeque.get(3)
+    const dataSize = dataSizes[socket.RCDeque.get(4)]
+    // If the length of the Deque is more than the header size and the size of the packet, make a buffer and then parse that buffer
+    if (socket.RCDeque.length >= 5 + dataCount * dataSize) {
+      // create another list to put the entire packet into
+      const packet = []
+      for (let i = 0; i < 5 + dataCount * dataSize; i++) {
```

Comment:
This has a lot of references to 5 (which is the header size). Maybe set this to a constant either at the top of rovecomm or in the manifest.

Reply:
Going to define the header length in the manifest instead of the rovecomm implementation to follow the design decisions which had us keep them separate to begin with.